### PR TITLE
[skip ci] Refactor arch

### PR DESCRIPTION
**Pull request by BlaineEXE**
Refactor ARCH to refer to Ceph's arch, not golang arch.

Convert ARCH to GO_ARCH where needed.
Ceph x86_64 -> golang amd64
Ceph aarch64 -> golang arm64

Built on changes from #932

### CODE REVIEW

**Comment by leseb** [comment]
@BlaineEXE Please rebase

**Comment by BlaineEXE** [comment]
@leseb Rebased

**Comment by ErwanAliasr1** [comment]
I don't get how you manage the translation between the x86_64 arch and the explicit need of having amd64 string in ARCH when using STAGE_REPLACE_WITH_ARCH like in  https://github.com/ceph/ceph-container/blob/master/ceph-releases/ALL/daemon/__WEB_INSTALL_ETCDCTL__

File: maint-lib/stagelib/envglobals.py [hunk 1]
```diff
@@ -88,3 +88,22 @@ def exportGitInfoEnvVars():
...
+
+
+def exportGoArchEnvVar():
+    """
+    Export the environment variable 'GO_ARCH' with the golang architecture equivalent to the
+    current Ceph arch. E.g., Ceph arch 'x86_64' equates to golang arch 'amd64'.
+    """
+    arch = getEnvVar('ARCH')
+    if arch in _CEPH_ARCH_TO_GOLANG_ARCH_CONVERSIONS:
+        os.environ['GO_ARCH'] = _CEPH_ARCH_TO_GOLANG_ARCH_CONVERSIONS[arch]
```
Comment (ErwanAliasr1) [inline maint-lib/stagelib/envglobals.py]: We surely need here to check in ARCH is part of _CEPH_ARCH_TO_GOLANG_ARCH_CONVERSIONS. If not (like a typo) that could trigger a fault here.

Comment (ErwanAliasr1) [reply maint-lib/stagelib/envglobals.py]: @BlaineEXE can you handle that so we can merge ?

Comment (ErwanAliasr1) [reply maint-lib/stagelib/envglobals.py]: @BlaineEXE perfectly true.... my bad.

**Pull request merged**